### PR TITLE
Data setup using cms

### PR DESCRIPTION
**Pull request by talwinder50**
Signed-off-by: talwinder.kaur <talwinder.kaur@securekey.com>

### CODE REVIEW

File: cmd/strapi-demo/main.go [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"time"
+)
+
+const adminURL = "http://localhost:1337/admin/auth/local/register"
```
Comment (fqutishat) [inline cmd/strapi-demo/main.go]: this values should be passed to main as arguments

File: scripts/strapi-setup.sh [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+$GENERATE_STUDENTAPI_COMMAND
+
+# generate the transcript api and model
+GENERATE_TRANSCRIPT_COMMAND="npx strapi generate:api transcripts StudentID:string Name:string University:string Course:string IssueDate:date Status:string TotalCredits:integer"
+
+$GENERATE_TRANSCRIPT_COMMAND
+
+cd ../../cmd/strapi-demo
+./main.go
```
Comment (fqutishat) [inline scripts/strapi-setup.sh]: you should build it

Comment (fqutishat) [reply scripts/strapi-setup.sh]: and pass location of binary to script

Comment (talwinder50) [reply scripts/strapi-setup.sh]: yes i fixed my local machine issue and script now is pointing to binary

File: cmd/strapi-demo/main.go (hunk 1 above)
Comment (talwinder50) [reply cmd/strapi-demo/main.go]: I have re-used cobra model for pass the parameters as the arguments for the consistency purpose.

**Comment by DRK3** [comment]
Remove the edge-auth-rest binary from the commit

File: cmd/strapi-demo/go.mod [hunk 3]
```diff
@@ -0,0 +1,21 @@
+// Copyright SecureKey Technologies Inc. All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+module github.com/trustbloc/edge-store/cmd/strapi-demo
+
+replace github.com/trustbloc/edge-store => ../..
+
+require (
+	github.com/konsorten/go-windows-terminal-sequences v1.0.2 // indirect
```
Comment (fqutishat) [inline cmd/strapi-demo/go.mod]: Why you need this dependency

File: cmd/strapi-demo/startcmd/start.go [hunk 4]
```diff
@@ -0,0 +1,218 @@
...
+func GetStartCmd() *cobra.Command {
+	startCmd := createStartCmd()
+
+	createFlags(startCmd)
+
+	return startCmd
+}
+func createStartCmd() *cobra.Command {
+	return &cobra.Command{
+		Use:   "start",
```
Comment (fqutishat) [inline cmd/strapi-demo/startcmd/start.go]: Change command from start

Comment (fqutishat) [reply cmd/strapi-demo/startcmd/start.go]: I suggest the command create-demo-data

File: scripts/strapi-setup.sh [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+GENERATE_STUDENTAPI_COMMAND="npx strapi generate:api studentcards StudentID:string Name:string University:string Semester:integer IssueDate:date"
+
+$GENERATE_STUDENTAPI_COMMAND
+
+# generate the transcript api and model
+GENERATE_TRANSCRIPT_COMMAND="npx strapi generate:api transcripts StudentID:string Name:string University:string Course:string IssueDate:date Status:string TotalCredits:integer"
+
+$GENERATE_TRANSCRIPT_COMMAND
+
+sleep 20s
```
Comment (fqutishat) [inline scripts/strapi-setup.sh]: Why sleep 20 seconds

Comment (talwinder50) [reply scripts/strapi-setup.sh]: when we create the schemas api in starpi, starpi gets restart as it  runs in the watch mode, the strapi demo binary need those schemas api (like studentcards, transcripts) to populate the data inside them.

File: cmd/strapi-demo/startcmd/start.go (hunk 4 above)
Comment (talwinder50) [reply cmd/strapi-demo/startcmd/start.go]: done

File: cmd/strapi-demo/go.mod (hunk 3 above)
Comment (talwinder50) [reply cmd/strapi-demo/go.mod]: Looks it was added somehow, i have deleted

File: cmd/strapi-demo/main.go [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+// For Demo you can verify the records by browsing /admin example : http://localhost:1337/admin/
+func main() {
+	rootCmd := &cobra.Command{
+		Use: "strapi-demo",
+		Run: func(cmd *cobra.Command, args []string) {
+			cmd.HelpFunc()(cmd, args)
+		},
+	}
+
+	rootCmd.AddCommand(startcmd.GetStartCmd())
```
Comment (fqutishat) [inline cmd/strapi-demo/main.go]: You need to change package name startcmd

Comment (talwinder50) [reply cmd/strapi-demo/main.go]: done

File: cmd/strapi-demo/createdemodata/start_test.go [hunk 7]
```diff
@@ -0,0 +1,141 @@
...
+	require.Equal(t, "", flag.Value.String())
+
+	flagAnnotations := flag.Annotations
+	require.Nil(t, flagAnnotations)
+}
+func TestAdminUserAndCreateRecordWithRoundTripper(t *testing.T) {
+	client := NewTestClient(func(req *http.Request) *http.Response {
+		// Test request parameters
+		return &http.Response{
+			StatusCode: 200,
```
Comment (DRK3) [inline cmd/strapi-demo/createdemodata/start_test.go]: Do you use the status code anywhere? I noticed that in the TestCreateAdminUserAndCreateRecordError function you don't have the status code defined. I'm wondering if you also don't need it here. Or if it should be used?

File: scripts/check_unit.sh [hunk 8]
```diff
@@ -40,3 +40,11 @@ PKGS=`go list github.com/trustbloc/edge-store/cmd/edge-auth-rest/... 2> /dev/nul
 go test $PKGS -count=1 -race -coverprofile=profile.out -covermode=atomic -timeout=10m
 amend_coverage_file
 cd "$pwd" || exit
+
+# Running edge-auth-rest unit tests
```
Comment (DRK3) [inline scripts/check_unit.sh]: Copy/paste error: "edge-auth-rest"

File: README.md [hunk 9]
```diff
@@ -6,3 +6,20 @@
 [![Build Status](https://dev.azure.com/trustbloc/edge/_apis/build/status/trustbloc.edge-store?branchName=master)](https://dev.azure.com/trustbloc/edge/_build/latest?definitionId=27&branchName=master)
 [![codecov](https://codecov.io/gh/trustbloc/edge-store/branch/master/graph/badge.svg)](https://codecov.io/gh/trustbloc/edge-store)
 [![Go Report Card](https://goreportcard.com/badge/github.com/trustbloc/edge-store)](https://goreportcard.com/report/github.com/trustbloc/edge-store)
+
+
+
+Create Demo Data using [Strapi](https://strapi.io/)
+###make strapi-start
```
[the remark this answers is not in the document]
Comment (talwinder50) [reply README.md]: done

File: README.md [hunk 10]
```diff
@@ -6,3 +6,20 @@
...
 [![Go Report Card](https://goreportcard.com/badge/github.com/trustbloc/edge-store)](https://goreportcard.com/report/github.com/trustbloc/edge-store)
+
+
+
+Create Demo Data using [Strapi](https://strapi.io/)
+###make strapi-start 
+Strapi start bring up the strapi on default 1337 port using mysql. 
+Once the strapi is up and running you can verify by browsing http://localhost:1337/admin/
+
+###make  strapi-setup
```
[the remark this answers is not in the document]
Comment (talwinder50) [reply README.md]: done

File: README.md [hunk 11]
```diff
@@ -6,3 +6,20 @@
...
+Strapi setup 
+1. Create the schemas APIs studentcards and transcripts 
+2. Create an admin user (username/password: strapi/strapi)
+3. Populate the schemas APIs with the data
+
+### Verify the demo data 
+1. Login into with the admin user credentials http://localhost:1337/admin/
```
[the remark this answers is not in the document]
Comment (talwinder50) [reply README.md]: done

File: cmd/strapi-demo/createdemodata/start.go [hunk 12]
```diff
@@ -0,0 +1,203 @@
...
+	createOrFetchRecord(client, authToken, parameters.adminURL+studentCardsEndpoint, "POST", studentRecord2)
+	createOrFetchRecord(client, authToken, parameters.adminURL+transcriptEndpoint, "POST", transcriptRecord1)
+	createOrFetchRecord(client, authToken, parameters.adminURL+transcriptEndpoint, "POST", transcriptRecord2)
+	createOrFetchRecord(client, authToken, parameters.adminURL+studentCardsEndpoint, "GET", nil)
+	createOrFetchRecord(client, authToken, parameters.adminURL+transcriptEndpoint, "GET", nil)
+
+	return nil
+}
+
+// createAdminUser create the admin user creates the user and generates the JWT token
```
[the remark this answers is not in the document]
Comment (talwinder50) [reply cmd/strapi-demo/createdemodata/start.go]: yea it was typo

File: cmd/strapi-demo/createdemodata/start.go [hunk 13]
```diff
@@ -0,0 +1,203 @@
...
+		}
+	}()
+
+	body, e := ioutil.ReadAll(resp.Body)
+
+	if e != nil {
+		return "", err
+	}
+
+	var adminUser map[string]interface{}
```
[the remark this answers is not in the document]
Comment (talwinder50) [reply cmd/strapi-demo/createdemodata/start.go]: strapi returns the jwt token and extra json infomation. 
Below is the strapi response for creating admin user. And i need Jwt token from this request. Therefore interface looks more appropriate to me. 
`{
    "jwt": "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpZCI6MTAsImlzQWRtaW4iOnRydWUsImlhdCI6MTU3NzQxNzU3NywiZXhwIjoxNTgwMDA5NTc3fQ.HCmhg8bxwJeriO7goKywU8L7St8I51E1AEaIC0M4v7E",
    "user": {
        "id": 10,
        "username": "strapi",
        "email": "user@strapi.io",
        "blocked": null,
        "isAdmin": true
    }
}`

File: cmd/strapi-demo/createdemodata/start_test.go (hunk 7 above)
Comment (talwinder50) [reply cmd/strapi-demo/createdemodata/start_test.go]: Not required removed

File: scripts/check_unit.sh (hunk 8 above)
Comment (talwinder50) [reply scripts/check_unit.sh]: fixed

File: cmd/strapi-demo/createdemodata/start.go (hunk 13 above)
[the remark this answers is not in the document]
Comment (DRK3) [reply cmd/strapi-demo/createdemodata/start.go]: Does the strapi response always have the same fields? If so, you can unmarshal to a custom type.  see DataVaultConfiguration in https://github.com/trustbloc/edge-store/blob/master/pkg/restapi/edv/operation/models.go

[the remark this answers is not in the document]
Comment (talwinder50) [reply cmd/strapi-demo/createdemodata/start.go]: i created custom type for user

**Pull request merged**